app/controllers/split_checkout_controller.rb

```diff
@@ -33,10 +34,14 @@ def update
     if confirm_order || update_order
       clear_invalid_payments
       advance_order_state
-      redirect_to_step
+      redirect_to current_step
     else
       flash.now[:error] = "Saving failed, please update the highlighted fields"
-      render :edit
+
+      render operations: cable_car.
+        replace("#checkout", partial("split_checkout/checkout")).
+        replace("#flashes", partial("shared/flashes", locals: { flashes: flash })),
+        status: :unprocessable_entity
```

Comment:
Here's the CableReady magic. I could have left this as just `render :edit` and it would have worked fine, but the whole page would have been rendered. Here we can now selectively re-render two _partials_ instead (the checkout form, which will have inline errors, and some flash messages).

:sparkles::sparkles:

Reply:
Nice one for sure. I should read some doc about cable ready. Do you have any advice?

Reply:
Normally CableReady works over Websockets, but they recently added a module (called "CableCar") which makes it work with Ajax requests as well.

The docs are really good: https://cableready.stimulusreflex.com/

There are **over 30** different CableReady operations, and I've only used one of them here (`replace`). It's basically a swiss-army-knife of server-initiated DOM manipulations.

Reply:
It also plays very nicely with **StimulusJS**, so if you add or update some HTML in the DOM with some new markup that has Stimulus controllers attached, those controllers will automatically initialize themselves (running anything in the `initialize()` or `connect()` methods), and any Stimulus controllers that are already on the page will not be interrupted when the DOM changes. This is quite different to other JS like jQuery, where unexpected changes to the DOM could either break something or require some code to be re-initialized manually. Know what I mean?

Reply:
> This is quite different to other JS like jQuery or Angular, where unexpected changes to the DOM's contents (outside of the normal full-page-request cycle) could either break something or require some code to be re-initialized manually

Yes, I can see this pain point (long time ago, then I switched to React :))